Quizzler-iOS13/Controller/ViewController.swift

```diff
@@ -26,35 +29,52 @@ class ViewController: UIViewController {
 
     @IBAction func answerButtonPressed(_ sender: UIButton) {
         
-        guard let userAnswer = sender.currentTitle else {
+        guard let selectedChoice = sender.currentTitle else {
             questionLabel?.text = "An unexpected error occured."
             assertionFailure("Unexpected case happened.")
             return
         }
         
-        let userGotItRight = quizBrain.checkAnswer(userAnswer: userAnswer)
+        let userGotItRight = quizBrain.checkAnswer(selectedChoice)
         
-        if userGotItRight {
+        if userGotItRight{
             sender.backgroundColor = UIColor.green
         } else {
             sender.backgroundColor = UIColor.red
+            showAlert(message: quizBrain.getExplanation())
         }
         
-        quizBrain.nextQuestion()
-        
+        quizBrain.moveToNextQuestion()
+     
         Timer.scheduledTimer(timeInterval: 0.2, target: self, selector: #selector(updateUI), userInfo: nil, repeats: false)
     }
     
     @objc func updateUI() {
         
         questionLabel.text = quizBrain.getQuestionText()
         
+        let currentQuestion = quizBrain.getCurrentQuestion()
+        
+        choice1.setTitle(currentQuestion?.choices[0], for: .normal)
+        choice2.setTitle(currentQuestion?.choices[1], for: .normal)
+        choice3.setTitle(currentQuestion?.choices[2], for: .normal)
+        
+        
         progressBar.progress = quizBrain.getProgress()
         scoreLabel.text = "Score: \(quizBrain.getScore())"
         
-        trueButton.backgroundColor = UIColor.clear
-        falseButton.backgroundColor = UIColor.clear
+        choice1.backgroundColor = UIColor.clear
+        choice2.backgroundColor = UIColor.clear
+        choice3.backgroundColor = UIColor.clear
+        
+        
     }
+    
+    func showAlert(message: String) {
+            let alert = UIAlertController(title: nil, message: message, preferredStyle: .alert)
+            alert.addAction(UIAlertAction(title: "Continue", style: .default, handler: nil))
+            present(alert, animated: true, completion: nil)
```

Comment:
ViewController should not handle navigation (same topic as we discuss before about Coordintators). It's fine to keep like this for now, but something to improve for the future

Reply:
We discussed the interactor pattern. Is that different from the coordinator pattern? 
Can you point me to an example using the coordinator pattern and how I should modify my code?
Here is the link I found:
https://www.hackingwithswift.com/articles/71/how-to-use-the-coordinator-pattern-in-ios-apps